instrumentation-genai/opentelemetry-instrumentation-google-genai/tests/generate_content/nonstreaming_base.py

```diff
@@ -92,6 +99,41 @@ def test_generated_span_has_minimal_genai_attributes(self):
             span.attributes["gen_ai.operation.name"], "generate_content"
         )
 
+    def test_span_and_event_still_written_when_response_is_exception(self):
+        self.configure_exception(ValueError("Uh oh!"))
+        patched_environ = patch.dict(
+            "os.environ",
+            {
+                "OTEL_INSTRUMENTATION_GENAI_CAPTURE_MESSAGE_CONTENT": "SPAN_AND_EVENT",
+                "OTEL_SEMCONV_STABILITY_OPT_IN": "gen_ai_latest_experimental",
+            },
+        )
+        with patched_environ:
+            _OpenTelemetrySemanticConventionStability._initialized = False
+            _OpenTelemetrySemanticConventionStability._initialize()
+            try:
+                self.generate_content(
+                    model="gemini-2.0-flash", contents="Does this work?"
+                )
+            except ValueError:
```

Comment:
https://docs.pytest.org/en/7.1.x/how-to/assert.html#assertions-about-expected-exceptions